### PR TITLE
[Minor] Add UT: UpdateSchemaWithComplexStruct

### PR DESCRIPTION
**Pull request by rominparekh**
**This PR:**
- Adds a unit test to update table schema with a complex (nested) struct
- Makes assertions on schema correctness

### CODE REVIEW

File: core/src/test/java/com/netflix/iceberg/hadoop/TestHadoopCommits.java [hunk 1]
```diff
@@ -242,4 +246,50 @@ public void testMergeAppend() throws Exception {
     Assert.assertEquals("Current snapshot should contain 1 merged manifest",
         1, metadata.currentSnapshot().manifests().size());
   }
+
+  @Test
+  public void testSchemaUpdateComplexType() throws Exception {
+    Assert.assertTrue("Should create v1 metadata",
+            version(1).exists() && version(1).isFile());
```
Comment (xabriel) [inline core/src/test/java/com/netflix/iceberg/hadoop/TestHadoopCommits.java]: Minor: indentation seems off compared to other code in file.

File: core/src/test/java/com/netflix/iceberg/hadoop/TestHadoopCommits.java [hunk 2]
```diff
@@ -242,4 +246,50 @@ public void testMergeAppend() throws Exception {
     Assert.assertEquals("Current snapshot should contain 1 merged manifest",
         1, metadata.currentSnapshot().manifests().size());
   }
+
+  @Test
+  public void testSchemaUpdateComplexType() throws Exception {
```
Comment (xabriel) [inline core/src/test/java/com/netflix/iceberg/hadoop/TestHadoopCommits.java]: Minor: How about you add this method right below existing `testSchemaUpdate()` test case? To me, they’re closely related.

File: core/src/test/java/com/netflix/iceberg/hadoop/TestHadoopCommits.java (hunk 1 above)
Comment (rominparekh) [reply core/src/test/java/com/netflix/iceberg/hadoop/TestHadoopCommits.java]: +1. Fixed indentation to 2 spaces.

File: core/src/test/java/com/netflix/iceberg/hadoop/TestHadoopCommits.java (hunk 2 above)
Comment (rominparekh) [reply core/src/test/java/com/netflix/iceberg/hadoop/TestHadoopCommits.java]: +1. Relocated test case below existing `testSchemaUpdate()`

**Comment by rominparekh** [comment]
Thanks for feedback @xabriel!

**Comment by rdblue** [comment]
@rominparekh, adding test cases is awesome!

**Pull request merged**